zhaquirks/tuya/ts0601_trv.py

```diff
@@ -1791,6 +1800,110 @@ def __init__(self, *args, **kwargs):
+            TuyaThermostatSystemMode.Off: Thermostat.SystemMode.Off,
+        }[x],
+        dp_converter=lambda x: {
+            Thermostat.SystemMode.Auto: TuyaThermostatSystemMode.Auto,
+            Thermostat.SystemMode.Heat: TuyaThermostatSystemMode.Heat,
+            Thermostat.SystemMode.Off: TuyaThermostatSystemMode.Off,
+        }[x],
+    )
+    .tuya_dp(
+        dp_id=2,
+        ep_attribute=TuyaThermostatV2.ep_attribute,
+        attribute_name=TuyaThermostatV2.AttributeDefs.occupied_heating_setpoint.name,
+        converter=lambda x: x * 10,
+        dp_converter=lambda x: x // 10,
+    )
+    .tuya_dp(
+        dp_id=3,
+        ep_attribute=TuyaThermostatV2.ep_attribute,
+        attribute_name=TuyaThermostatV2.AttributeDefs.local_temperature.name,
+        converter=lambda x: x * 10,
+    )
+    .tuya_dp(
+        dp_id=6,
+        ep_attribute=TuyaThermostatV2.ep_attribute,
+        attribute_name=TuyaThermostatV2.AttributeDefs.running_state.name,
+        converter=lambda x: 0x01 if not x else 0x00,  # Heat, Idle
+    )
+    .tuya_binary_sensor(
+        dp_id=7,
+        attribute_name="window_open",
+        device_class=BinarySensorDeviceClass.WINDOW,
+        translation_key="window_open",
+        fallback_name="Window open",
+    )
+    .tuya_switch(
+        dp_id=8,
+        attribute_name="window_detection",
+        translation_key="window_detection",
+        fallback_name="Open window detection",
+    )
+    .tuya_switch(
+        dp_id=12,
+        attribute_name="child_lock",
+        translation_key="child_lock",
+        fallback_name="Child lock",
+    )
+    .tuya_battery(dp_id=13)
+    .tuya_binary_sensor(
+        dp_id=14,
+        attribute_name="error_or_battery_low",
+        translation_key="error_or_battery_low",
+        fallback_name="Error or battery low",
+    )
+    .tuya_number(
+        dp_id=15,
+        attribute_name="min_temperature",
+        type=t.uint16_t,
+        min_value=1,
+        max_value=15,
+        unit=UnitOfTemperature.CELSIUS,
+        step=1,
+        translation_key="min_temperature",
+        fallback_name="Min temperature",
+    )
+    .tuya_number(
+        dp_id=16,
+        attribute_name="max_temperature",
+        type=t.uint16_t,
+        min_value=15,
+        max_value=35,
+        unit=UnitOfTemperature.CELSIUS,
+        step=1,
+        translation_key="max_temperature",
+        fallback_name="Max temperature",
+    )
+    .tuya_dp(
+        dp_id=101,
+        ep_attribute=TuyaThermostatV2.ep_attribute,
+        attribute_name=TuyaThermostatV2.AttributeDefs.local_temperature_calibration.name,
+        converter=lambda x: x,
+        dp_converter=lambda x: x + 0x100000000 if x < 0 else x,
+    )
+    .tuya_enum(
+        dp_id=114,
+        attribute_name="eco_mode",
+        enum_class=TuyaThermostatEcoMode,
+        translation_key="eco_mode",
+        fallback_name="Eco mode",
+    )
```

Comment:
Comfort and eco might need to be two presets for the climate entity, which we unfortunately can't do with quirks v2 yet. I'm not sure if we're allowed to (or should) add a separate select entity for this.
Might have to come back to this.

Reply:
Ok, we already have some TRV quirks that have the preset mode as another option.
There are also some older entities in ZHA directly that expose presets as a select entity. Z2M does the same for a lot of TRVs too.
To get these devices working at all, we can have this separate for now. In the future, we should look at making the thermostat platform "quirk v2" compatible.